### PR TITLE
Added support for loading a watermask for Quantized-Mesh terrain tiles.

### PR DESCRIPTION
**Pull request by abwood**
These changes add support for the new quantized-mesh extension, watermask.  This extension is meant to be a drop-in replacement for the heightmap-1.0 terrain watermask, and shares the same format; 1 byte if all land/water, 64k mask for tiles with a mix of land and water.

Related changes to the Quantized-Mesh 1.0 spec:
https://github.com/AnalyticalGraphicsInc/cesium-assets-list/pull/6

Updates to the Cesium Terrain Tutorial, to be reviewed and merged in after this PR:
https://github.com/AnalyticalGraphicsInc/cesium-website/tree/terraintutorial-watermaskupdate

The world tileset has been deployed along with a new terrain server install. Check out the Terrain SandCastle example to see the water effect in action and compare with the small terrain watermask.

### CODE REVIEW

File: Source/Core/CesiumTerrainProvider.js [hunk 1]
```diff
@@ -57,6 +57,7 @@ define([
      * @param {String} options.url The URL of the Cesium terrain server.
      * @param {Proxy} [options.proxy] A proxy to use for requests. This object is expected to have a getURL function which returns the proxied URL, if needed.
      * @param {Boolean} [options.requestVertexNormals=false] Flag that indicates if the client should request additional lighting information from the server, in the form of per vertex normals if available.
+     * @param {Boolean} [options.requestWaterMask=false] Flag that indicates if the client should request per tile water masks from the server,  if available.
```
Comment (pjcozzi) [inline Source/Core/CesiumTerrainProvider.js]: Add to this 1.7 in CHANGES.md.

File: Specs/Core/CesiumTerrainProviderSpec.js [hunk 2]
```diff
@@ -63,10 +63,34 @@ defineSuite([
         return returnTileJson('Data/CesiumTerrainTileJson/OctVertexNormals.tile.json');
     }
 
+    function returnWaterMaskTileJson() {
+        return returnTileJson('Data/CesiumTerrainTileJson/WaterMask.tile.json');
```
Comment (pjcozzi) [inline Specs/Core/CesiumTerrainProviderSpec.js]: Do we have tests for both cases: all water or land, and a mix?

**Comment by pjcozzi** [comment]
Can we update the Sandcastle terrain example?  Perhaps drop `smallterrain` and then add checkboxes or whatever for lighting and water.

**Comment by abwood** [comment]
@pjcozzi The new tileset is deployed on cesiumjs.org/stk-terrain and assets.agi.com/stk-terrain.  I've updated the Terrain Sandcastle (and cesium inspector) to demonstrate these new features.  This newly generated watermask is a bit smoother along coastlines when compared to `smallterrain` so I left it in for comparison.

**Comment by pjcozzi** [comment]
I would still drop `smallterrain` from Sandcastle and any reference documentation examples.  We should show only relevant forward-facing examples.  We can put a screenshot comparison in a blog post.

**Comment by mramato** [comment]
:+1: on getting rid of smallterrain

**Comment by abwood** [comment]
While i'm in there, should I drop it from the BaseLayerPicker as well?

**Comment by mramato** [comment]
Yes, I would remove all traces of it from Cesium.  I would also change it so the default STK Terrain provider for the base layer picker always gets normals and water by default.

**Comment by abwood** [comment]
Ok, I have removed references of smallterrain everywhere except the GlobeSurfaceTileSpec.  

I think this is a separate discussion, but the GlobeSurfaceTileSpec tests assume that all upsampling of child tiles will happen immediately after the root tile is loaded.  Since the upsampling of tiles is done in asynchronously via webworkers for Quantized-mesh, everything falls apart. e.g., In the test, `loaded terrain replacing upsampled terrain triggers re-upsampling and re-loading of children`, root.loadedTerrain becomes undefined once the tile is published, all while childTile and grandchildTile are upsampling. CC @kring

**Comment by kring** [comment]
> GlobeSurfaceTileSpec tests assume that all upsampling of child tiles will happen immediately after the root tile is loaded

Yeah, the existing tests will need significant changes to work with STK World Terrain, sorry.

**Comment by abwood** [comment]
It's no big deal.  I don't think we need to rush to fix those tests as they are not meant to be tied to any particular tileset.  I just wanted to mention it so that it wasn't seen as an oversight.

**Comment by pjcozzi** [comment]
Thanks @abwood.  This looks good.

**Pull request merged**